python/tests/test_1_online.py

```diff
@@ -2055,6 +2055,36 @@ def test_delete_deltachat_folder(acfactory):
     assert "DeltaChat" in ac1.direct_imap.list_folders()
 
 
+def test_aeap_flow_unverified(acfactory, lp):
+    """Test that a new address is added to a contact when it changes its address."""
+    ac1, ac2, ac1new = acfactory.get_online_accounts(3)
+    chat = acfactory.get_accepted_chat(ac1, ac2)
+
+    lp.sec("sending first message")
+    msg_out = chat.send_text("old address")
+
+    lp.sec("receiving first message")
+    msg_in_1 = ac2._evtracker.wait_next_incoming_message()
+    assert msg_in_1.text == msg_out.text
+
+    lp.sec("changing email account")
+    ac1.set_config("addr", ac1new.get_config("addr"))
+    ac1.set_config("mail_pw", ac1new.get_config("mail_pw"))
+    ac1.stop_io()
+    configtracker = ac1.configure()
+    configtracker.wait_finish()
+    ac1.start_io()
+
+    lp.sec("sending second message")
+    msg_out = chat.send_text("changed address")
+
+    lp.sec("receiving second message")
+    msg_in_2 = ac2._evtracker.wait_next_incoming_message()
+    assert msg_in_2.text == msg_out.text
+    #assert msg_in_2.chat.id == msg_in_1.chat.id
+    assert msg_in_2.get_sender_contact() == msg_in_1.get_sender_contact()
```

Comment:
I wrote a new test, but it's failing. Am I checking correctly whether the transition worked?

Reply:
No, you should either:
- create a group before, and test whether the old contact was replaced with the new contact in in the group
- Check whether the AEAP transition info message was added

check_that_transition_worked() in src/tests/aeap.rs does both.

Also, I just noticed, since transitions are only done for verified contacts/groups for now for security reasons, the two accounts have to verify each other first; maybe you can extend test_qr_verified_group_and_chatting().

Reply:
But again, if it's too much hassle, it wouldn't be the only feature that's only tested by Rust, not Python tests, so it should be fine to not test it here I think (although of course the bindings would be untested then).

Reply:
Hm, shouldn't these two contacts be the same? https://github.com/deltachat/deltachat-core-rust/runs/7689203037?check_suite_focus=true

Reply:
If not, we can just take out the last commit - the ["use a verified group for the AEAP test"](https://github.com/deltachat/deltachat-core-rust/pull/3530/commits/6aba9f3ca9a2ed793ab085163f457d6173d25bb6) commit already proved that AEAP kind of works, maybe we don't need the check introduced by ["tests: ensure that contact of ac1 stays the same"](https://github.com/deltachat/deltachat-core-rust/pull/3530/commits/0aaa33a0d2fd75ff6297ba681ed9de4127bbc995), which throws the error